### PR TITLE
Session auth

### PR DESCRIPTION
**Pull request by Iamayushgupta**

### CODE REVIEW

File: controller/authController.go [hunk 1]
```diff
@@ -6,42 +6,125 @@ import (
...
+		c.JSON(http.StatusBadRequest, gin.H{"error": "Invalid input"})
 		return
 	}
 
 	err := user.SignUp(config.DB)
 	if err != nil {
-		log.Printf("Failed to sign up user: %v", err)
-		c.JSON(http.StatusInternalServerError, gin.H{"error": "failed to sign up"})
+		// Sends back a JSON response
+		c.JSON(500, gin.H{"error": err})
```
Comment (abhigupta4) [inline controller/authController.go:27]: Why did you change this?

File: controller/authController.go [hunk 2]
```diff
@@ -6,42 +6,125 @@ import (
...
+		c.JSON(http.StatusBadRequest, gin.H{"error": "Invalid input"})
 		return
 	}
 
 	err := user.Login(config.DB)
+
 	if err != nil {
-		log.Printf("Login failed for user %s: %v", user.Username, err)
-		c.JSON(http.StatusUnauthorized, gin.H{"error": "incorrect username or password"})
+		if err.Error() == "user not found"{
```
Comment (abhigupta4) [inline controller/authController.go:47]: We discussed to change this, right?

File: controller/authController.go (hunk 1 above)
Comment (Iamayushgupta) [reply controller/authController.go:27]: I logging this error in User Model itself

File: controller/authController.go (hunk 2 above)
Comment (Iamayushgupta) [reply controller/authController.go:47]: Yes, actually main isse purra change karne vala tha but phir kuch error aa raha tha which i couldnt resolve, i will do it later, because abhi code main redundancy bhi hain
